scripts/dist-fix.js

```diff
@@ -1,70 +1,26 @@
-const {
-  mkdir,
-  readdir,
-  rename,
-  rm,
-  writeFile,
-  copyFile,
-  readFile,
-  unlink,
-} = require('node:fs/promises')
+// @ts-check
+const {readdir, writeFile, readFile} = require('node:fs/promises')
 const path = require('node:path')
 
 ;(async () => {
   const distPath = path.join(__dirname, '../dist')
-  const distCjsPath = path.join(distPath, 'cjs')
-  const distEsmPath = path.join(distPath, 'esm')
-
-  const [dist, distEsm] = await Promise.all([
-    readdir(distPath),
-    readdir(distEsmPath),
-    rm(distCjsPath, {force: true, recursive: true}),
-  ])
+  const dist = await readdir(distPath)
 
+  // Inject type declarations for deno.
+  // See https://docs.deno.com/runtime/manual/advanced/typescript/types
   await Promise.all([
-    mkdir(distCjsPath),
-    writeFile(
-      path.join(distEsmPath, 'package.json'),
-      JSON.stringify({type: 'module', sideEffects: false}),
-    ),
     ...dist
-      .filter((distFilePath) => distFilePath.match(/\.d\.ts$/))
-      .map((distFilePath) =>
-        copyFile(
-          path.join(distPath, distFilePath),
-          path.join(distEsmPath, distFilePath),
-        ),
-      ),
-    ...distEsm
-      .filter((esmFilePath) => esmFilePath.match(/\.js$/))
-      .map(async (esmFilePath) => {
-        const distEsmFilePath = path.join(distEsmPath, esmFilePath)
-
-        const esmFile = await readFile(distEsmFilePath)
-        const esmFileContents = esmFile.toString()
-
-        const dtsFilePath = `./${esmFilePath.replace('.js', '.d.ts')}`
-
-        const denoFriendlyEsmFileContents = [
-          `/// <reference types="${dtsFilePath}" />`,
-          esmFileContents,
-        ].join('\n')
+      .filter((filePath) => filePath.match(/\.m?js$/))
+      .filter((filePath) => !filePath.startsWith('chunk-'))
+      .map(async (filename) => {
+        const distFilePath = path.join(distPath, filename)
 
-        await unlink(distEsmFilePath)
+        const dtsFilePath = filename.replace(/\.(m?)js$/, '.d.$1ts')
+        const denoFriendlyJsFileContents =
+          `/// <reference types="${dtsFilePath}" />\n` +
```

Comment:
let's add a minimal deno test @ `tests/deno` to verify we're building this stuff correctly..
see https://github.com/kysely-org/kysely/tree/master/test/deno for reference (just local is fine).

Reply:
added

disclaimer: never used deno, so very new to both it and this kind of deno testing. 😅 tried to roughly replicate the setup from the kysely repo with a new bare-bones test. not sure if this aligns with what you had in mind though(?)

I also don't think this setup actually tests our TS-only `/// <reference ...>` injection at all. If I alter the reference, the deno script still completes fine. So this might only somewhat tests if the bundled runtime code is deno-compatible.